src/Controls/src/Core/ElementTemplate.cs

```diff
@@ -21,7 +21,14 @@ internal ElementTemplate(Type type) : this()
 
 			_canRecycle = true;
 
-			LoadTemplate = () => Activator.CreateInstance(type);
+			LoadTemplate = () =>
+			{
+				if (Application.Current?.Handler?.MauiContext?.Services != null)
```

Comment:
Services need to be retrieved from the context the element is being used in. The services all get scoped differently based on the window/modal they are apart of.   This it the part I'm hoping doesn't become too tricky. In theory you can find the correct services via the parent

`_parent.FindMauiContext().Services`

Reply:
Yeah, I'm starting to bump into the trickiness here. The ShellContent's ContentTemplate's _parent is null when LoadTemplate() is being called. The ShellContent.Parent does exist though, maybe I can either try to set ContentTemplate.Parent with "this" (ShellContent) when GetOrCreateContent() is being called, or something to that effect? Or is that a bad idea?

Reply:
@PureWeen does this seem too hacky to you?

Reply:
Looking at `ElementTemplate` a bit closer I think that it was incorrectly made to implement `IElement` eight or so years ago when that structure was first created. AFAICT the parent is never set on `ElementTemplate` anywhere ever.  Can we utilize the fact that the setter for `ElementTemplate.LoadTemplate` is public?  Then inside ShellContent we just set our own `LoadTemplate` ? I think I'm going to walk back my statement about this being a `DataTemplate` enhancement :-)  I think the use of a `ServiceProvider` with a `DataTemplate` should probably be isolated to `Shell`. The only other cases where we use `DataTemplates` are with `ListView/CollectionView` which don't really seem to fit the bill here and we probably don't want to add the additional overhead to those creators.

Reply:
@PureWeen I don't think it's that simple, because of how `DataTemplate` in the XAML gets instantiated, in `DataTemplateExtension`. The constructor takes a `Type` which gets used by `LoadTemplate`. We'd have to somehow get the type into whatever we set for `Loadtemplate` in `ShellContent`. 

https://github.com/dotnet/maui/blob/18e0f4ebbcca7c904ccdb67e6439cb72382e2a40/src/Controls/src/Xaml/MarkupExtensions/DataTemplateExtension.cs#L22-L25

Reply:
Hunh, that feels like a sort of useless Func :-/ It just uses a closure inside the `ctor`  👎 What if you just store the type locally and expose it out using an internal? does that give you what you need?

Reply:
@PureWeen Yes, this seems to work.
Like you say, this is limited to `DataTemplate` in `ShellContent`, not working with `DataTemplate` in general.